packages/features/auth/lib/next-auth-options.ts

```diff
@@ -115,7 +115,7 @@ const providers: Provider[] = [
         throw new Error(ErrorCode.IncorrectEmailPassword);
       }
 
-      if (user.password || !credentials.totpCode) {
+      if (user.password && !credentials.totpCode) {
```

Comment:
We should check if password is correct when signing in when user has provided user and no TOTP code, otherwise treat this as a 2FA login